yb-voyager/src/tgtdb/yugabytedb_test.go

```diff
@@ -192,3 +192,55 @@ func TestYugabyteGetNonEmptyTables(t *testing.T) {
 	log.Infof("non empty tables: %+v\n", actualTables)
 	testutils.AssertEqualNameTuplesSlice(t, expectedTables, actualTables)
 }
+
+func TestGetPrimaryKeyConstraintName(t *testing.T) {
+	testYugabyteDBTarget.ExecuteSqls(
+		`CREATE SCHEMA test_schema;`,
+		`CREATE TABLE test_schema.foo (
+			id INT PRIMARY KEY,
+			name TEXT
+		);`,
+		`CREATE TABLE test_schema.bar (
+			id INT,
+			name TEXT,
+			CONSTRAINT bar_primary_key PRIMARY KEY (id)
+		);`,
+		`CREATE TABLE test_schema.baz (
+			id INT,
+			name TEXT
+		);`,
+		`CREATE TABLE test_schema."CASE_sensitive" (
+			id INT,
+			name TEXT,
+			CONSTRAINT "CASE_sensitive_pkey" PRIMARY KEY (id)
+		);`,
+	)
+	defer testYugabyteDBTarget.ExecuteSqls(`DROP SCHEMA test_schema CASCADE;`)
+
+	tests := []struct {
+		table          sqlname.NameTuple
+		expectedPKName string
+	}{
+		{
+			table:          sqlname.NameTuple{CurrentName: sqlname.NewObjectName(POSTGRESQL, "test_schema", "test_schema", "foo")},
+			expectedPKName: "foo_pkey",
+		},
+		{
+			table:          sqlname.NameTuple{CurrentName: sqlname.NewObjectName(POSTGRESQL, "test_schema", "test_schema", "bar")},
+			expectedPKName: "bar_primary_key",
+		},
+		{
+			table:          sqlname.NameTuple{CurrentName: sqlname.NewObjectName(POSTGRESQL, "test_schema", "test_schema", "baz")},
+			expectedPKName: "",
+		},
+		{
+			table:          sqlname.NameTuple{CurrentName: sqlname.NewObjectName(POSTGRESQL, "test_schema", "test_schema", "CASE_sensitive")},
+			expectedPKName: "CASE_sensitive_pkey",
```

Comment:
nice. Have we verified that this appears in this exact manner in the error message?

Reply:
yes, case senstivity is covered as part of `VIOLATES_UNIQUE_CONSTRAINT_ERROR_RETRYABLE_FAST_PATH` error detection logic